tools/integration_tests/util/client/control_client.go

```diff
@@ -19,13 +19,45 @@ import (
 	"fmt"
 	"log"
 	"strings"
+	"time"
 
 	control "cloud.google.com/go/storage/control/apiv2"
 	"cloud.google.com/go/storage/control/apiv2/controlpb"
+	"github.com/googleapis/gax-go/v2"
+	"google.golang.org/grpc/codes"
 )
 
+func storageControlClientRetryOptions() []gax.CallOption {
+	return []gax.CallOption{
+		gax.WithTimeout(300000 * time.Millisecond),
+		gax.WithRetry(func() gax.Retryer {
+			return gax.OnCodes([]codes.Code{
+				codes.ResourceExhausted,
+				codes.Unavailable,
+				codes.DeadlineExceeded,
+				codes.Internal,
+				codes.Unknown,
+			}, gax.Backoff{
+				Max:        30 * time.Second,
+				Multiplier: 2,
+			})
+		}),
+	}
+}
+
 func CreateControlClient(ctx context.Context) (client *control.StorageControlClient, err error) {
 	client, err = control.NewStorageControlClient(ctx)
+
+	client.CallOptions.RenameFolder = storageControlClientRetryOptions()
+	client.CallOptions.GetFolder = storageControlClientRetryOptions()
+	client.CallOptions.GetStorageLayout = storageControlClientRetryOptions()
+	client.CallOptions.CreateFolder = storageControlClientRetryOptions()
+	client.CallOptions.DeleteFolder = storageControlClientRetryOptions()
+	client.CallOptions.CreateManagedFolder = storageControlClientRetryOptions()
+	client.CallOptions.DeleteManagedFolder = storageControlClientRetryOptions()
```

Comment:
Lets keep the scope to only Create and Delete Managed folders which are needed here. The others we can remove. That way we keep this only for cases where the workflow cannot be invoked by gcsfuse.

Reply:
Okay.. as and when we add other methods like Create, DeleteFolder we can make it retryable at that time. Done.

Reply:
Ideally create, delete folders should not come via this route. They are valid workflows in gcsfuse so its better they are invoked that way so they are tested as well.

Reply:
Yeah, but sometimes we write test cases, where we create/delete data from outside GCSFuse(e.g. Storage client, gcloud) and validate the result with GCSFuse. At that point we need this.

Reply:
Sure, yea we can take it case by case basis.